data/transactions/logic/assembler_test.go

```diff
@@ -529,28 +553,37 @@ func testProg(t testing.TB, source string, ver uint64, expected ...Expect) *OpSt
 	t.Helper()
 	program := strings.ReplaceAll(source, ";", "\n")
 	ops, err := AssembleStringWithVersion(program, ver)
+	semiOps, semiErr := AssembleStringWithVersion(addSemis(program), ver)
```

Comment:
I think running all our tests through this weird semicolon treatment makes it really hard to know if we've got them handled exactly right. I'd rather see a PR with just semi-colon handling, and all the tests fixed. I know that a lot of `Expect` changes.

But we need to know that that we've properly tested all sorts of semi colon funny business, and we won't be very confident if the tests have all been manipulated this way.

Reply:
Agreed, this was just to give some confidence it's working. The best solution I've come up with is to change linerror to also have a field that tracks which statement in the line it was from, and then make sure that together the line number and statement number line up with the expected number. This would also let us be more precise in our erroring when people use semicolons.